python/lsst/afw/math/backgroundList.py

```diff
@@ -164,64 +164,47 @@ def readFits(fileName, hdu=0, flags=0):
 
         self = BackgroundList()
 
-        if hdu == DEFAULT_HDU:
-            hdu = -1
-        else:
-            # we want to start at 0 (post RFC-304), but are about to increment
-            hdu -= 1
-
-        fits = Fits(fileName, "r")
-        fits.setHdu(hdu + 1)
-        if fits.checkCompressedImagePhu():
-            hdu += 1
-
-        while True:
-            hdu += 1
-
-            md = dafBase.PropertyList()
-            try:
-                img = afwImage.ImageF(fileName, hdu, md)
-                hdu += 1
-            except FitsError:
-                break
-
-            msk = afwImage.Mask(fileName, hdu)
-            hdu += 1
-            var = afwImage.ImageF(fileName, hdu)
-
-            statsImage = afwImage.makeMaskedImage(img, msk, var)
-
-            x0 = md.getScalar("BKGD_X0")
-            y0 = md.getScalar("BKGD_Y0")
-            width = md.getScalar("BKGD_WIDTH")
-            height = md.getScalar("BKGD_HEIGHT")
-            imageBBox = lsst.geom.BoxI(lsst.geom.PointI(
-                x0, y0), lsst.geom.ExtentI(width, height))
-
-            interpStyle = afwMath.Interpolate.Style(md.getScalar("INTERPSTYLE"))
-            undersampleStyle = afwMath.UndersampleStyle(
-                md.getScalar("UNDERSAMPLESTYLE"))
+        f = Fits(fileName, 'r')
+        nHdus = f.countHdus()
+        f.closeFile()
+        if nHdus % 3 != 0:
+            raise RuntimeError(f"BackgroundList FITS file {fileName} has {nHdus} HDUs;"
+                               f"expected a multiple of 3 (compression is not supported).")
```

Comment:
Is the `f`-string needed on this line?

Reply:
No, I just like to use or not use `f`-strings consistently when I'm relying on multi-line string concatenation, so changing the line break can't break things.